File: leet_easy_c++/leet_136/leet_136b.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> umap;
        int len = nums.size();
        if(len == 0) return 0;
        for(int i=0; i<len; i++)  umap[nums[i]]=0;
        for(int i=0; i<len; i++){           
            if(umap[nums[i]]==1) umap.erase(nums[i]);
            else  umap[nums[i]]++;
        }  
        return umap.begin()->first;
    }
};

File: leet_easy_java/leet_136/leet_136b.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> hmap = new HashMap<Integer, Integer>();
        for(int x : nums) hmap.put(x, ( hmap.get(x)==null ? 1 : 2));
        for (Map.Entry<Integer, Integer> entry : hmap.entrySet()) {
            if(entry.getValue() == 1)
                return entry.getKey();
        }
        return -1; //~ if reached here --> would signal an error
    }
}